### PR TITLE
Release 0.4.1 & fix state preparation in remote backend

### PR DESCRIPTION
**Pull request by jrj-d**

### CODE REVIEW

File: tests/test_translation_plugin.py [hunk 1]
```diff
@@ -1,170 +1,90 @@
-##############################################################################
```
Comment (MaxleDrut) [inline tests/test_translation_plugin.py:1]: Could you please put this banner back?

File: tests/test_translation_plugin.py [hunk 2]
```diff
@@ -1,170 +1,90 @@
...
-
-def test_complex_unsupported() -> None:
-    c = QuantumCircuit(2, 1)
-    state = np.array([1, -3j])
-    state /= np.sqrt(np.sum(state * np.conj(state)))
-    c.initialize(state, 0)
-    with pytest.raises(AliceBobTranspilationException):
-        _pm.run(c)
+
+def _check_initialize(circuit: QuantumCircuit, expected: str) -> None:
```
Comment (MaxleDrut) [inline tests/test_translation_plugin.py]: I'd say we could have a more explicit naming of this function and the one below.
How about `_assert_one_initialize` and `_assert_many_initializes`? (assert_one and assert_many seems to be the common structure in the Python test modules)

File: tests/test_translation_plugin.py [hunk 3]
```diff
@@ -1,170 +1,90 @@
...
+    _check_initialize(transpiled, '+0')
+
+    # a state vector should fail
+    circ = QuantumCircuit(2)
+    circ.initialize([1, 0, 0, 0])
+    with pytest.raises(TranspilerError):
+        pm.run(circ)
+
+
+def _check_initializes(circuit: QuantumCircuit, expected: List[str]) -> None:
```
Comment (MaxleDrut) [inline tests/test_translation_plugin.py]: nit: wdyt of moving this function at beginning of the file, just after _check_initialize?

File: qiskit_alice_bob_provider/local/translation_plugin.py [hunk 4]
```diff
@@ -262,7 +115,7 @@ def pass_manager(
         }
 
         # Use as basis the same pass manager as the LocalStatePreparationPlugin
-        pm = LocalStatePreparationPlugin().pass_manager(
+        pm = StatePreparationPlugin().pass_manager(
             pass_manager_config=pass_manager_config,
             optimization_level=optimization_level,
         )
```
Comment (MaxleDrut) [inline qiskit_alice_bob_provider/local/translation_plugin.py]: I find it confusing to have a shared `translation_plugin` for local and remote and a second one with the same name specifically for local. Could we think of a different name for this file like `local_translation.py`?

Comment (MaxleDrut) [reply qiskit_alice_bob_provider/local/translation_plugin.py]: My bad this is addressed in a later commit actually, nice :+1:

**Review by MaxleDrut — COMMENTED** [review]
Overall this looks good to me, great improvement!

File: tests/test_translation_plugin.py (hunk 1 above)
Comment (jrj-d) [reply tests/test_translation_plugin.py:1]: This file was deleted, can you please resolve?

Comment (MaxleDrut) [reply tests/test_translation_plugin.py:1]: Yes

File: tests/test_translation_plugin.py (hunk 2 above)
Comment (jrj-d) [reply tests/test_translation_plugin.py]: Done!

File: tests/test_translation_plugin.py (hunk 3 above)
Comment (jrj-d) [reply tests/test_translation_plugin.py]: Done!

**Review by MaxleDrut — APPROVED** [review]
:rocket:

**Pull request merged**